pkg/kubelet/volume/plugins.go

```diff
@@ -80,6 +80,17 @@ type Host interface {
 
 	// GetKubeClient returns a client interface
 	GetKubeClient() client.Interface
+
+	// NewWrapperBuilder finds an appropriate plugin with which to handle
+	// the provided spec.  This is used to implement volume plugins which
+	// "wrap" other plugins.  For example, the "secret" volume is
+	// implemented in terms of the "emptyDir" volume.
+	NewWrapperBuilder(spec *api.Volume, podUID types.UID) (Builder, error)
```

Comment:
@thockin This won't work for me as-is.   The introduction of PersistentVolumeSource means my volume can no longer ape being another volume.  

Volume -> VolumeSource
PersistentVolume -> PersistentVolumeSource

I can still fake it within my own volume, but I'm pretty sure your intent was to keep my volume ignorant of all others.  Faking it also means the volume (say, GCEPD) needs to be present in both VolumeSource and PersistentVolumeSource.  I'm sure the intent was to remove volumes from VolumeSource.

I think we need an interface for both Volume and PersistentVolume to implement, so we can treat them uniformly.